src/index.ts

```diff
@@ -36,6 +36,29 @@ const listen =
       type: "LISTEN",
     });
 
+interface Grammar {
+    entities: {
+      [index: string]: string;
+        };
+      }
+
+const grammar = {
```

Comment:
This needs some extra explanation. How it is going to scale? Now it seems like a semantic interpretation of "put pink lamp on the shelf" rather than something that would work for other cases. How would your code change when you add more colours, object and places?